htdocs/compta/accounting-files.php

```diff
@@ -214,9 +214,9 @@
 			if (!empty($sql)) {
 				$sql .= " UNION ALL";
 			}
-			$sql .= " SELECT t.rowid as id, t.entity, t.label as ref, 1 as paid, amount as total_ht, amount as total_ttc, 0 as total_vat, '".$db->escape($conf->currency)."' as currency, t.fk_user as fk_soc, t.datep as date, t.dateep as date_due, 'SalaryPayment' as item, CONCAT(CONCAT(u.lastname, ' '), u.firstname)  as thirdparty_name, '' as thirdparty_code, c.code as country_code, '' as vatnum, ".PAY_DEBIT." as sens";
-			$sql .= " FROM ".MAIN_DB_PREFIX."payment_salary as t LEFT JOIN ".MAIN_DB_PREFIX."user as u ON u.rowid = t.fk_user LEFT JOIN ".MAIN_DB_PREFIX."c_country as c ON c.rowid = u.fk_country";
-			$sql .= " WHERE datep between ".$wheretail;
+			$sql .= " SELECT t.rowid as id, t.entity, t.label as ref, paye as paid, amount as total_ht, amount as total_ttc, 0 as total_vat, '".$db->escape($conf->currency)."' as currency, t.fk_user as fk_soc, t.datesp as date, t.dateep as date_due, 'SalaryPayment' as item, CONCAT(CONCAT(u.lastname, ' '), u.firstname)  as thirdparty_name, '' as thirdparty_code, c.code as country_code, '' as vatnum, ".PAY_DEBIT." as sens";
+			$sql .= " FROM ".MAIN_DB_PREFIX."salary as t LEFT JOIN ".MAIN_DB_PREFIX."user as u ON u.rowid = t.fk_user LEFT JOIN ".MAIN_DB_PREFIX."c_country as c ON c.rowid = u.fk_country";
```

Comment:
A salary can be payed in 2 times. First an advance on day x in cash then the rest on day y in check, for example.
So 1 salary can have 2 payments at 2 different dates.

So the question is what do we want to export ?

A) Here the current design is to export ithe detail of payments. So with example, it will return 2 lines.
The table for payment of salaries is payment_salary and if we change the table, it means we do not extract the same thing, we will get one line that is the salary. Why not, it is a different feature.
If what we want is this case A, if there is a problem into the link to salary or payment record, this can be fixed by getting the good id of salary from the payment.

B) But if we want to export the salary (1 line) and not the payments (2 lines), it is a different feature. In such a case, we must also modify the label of the export. Instead of calling it "Payments of salary", it must be called "Salaries". We must also know that in this case, the date and the amount in salary is not the real amount. It is just the date of start of period for the salary and the expected amount to pay. 

So:
- If what we want in the file is to list of paymet (so the case A), we must fix by adding code to get the ID of salary from the ID of salary to show correcly the good link to document (the original bug).
- if what we want if the file is the list of expected payment of salary (so the case B) , we must also fix this everywhere by changing also the label. The key 'selectpaymentsofsalaries' should also be named 'selectsalaries' instead to avoid confusion.

Reply:
Interesting return. I'll think about it and get back to you.